guides/theming.md

```diff
@@ -1,311 +1,359 @@
+   100: rgba(black, 0.87),
+   200: rgba(black, 0.87),
+   300: white,
+   // ... continues to 900
+ )
+);
 ```
 
-Alternatively, you can just reference the file directly. This would look something like:
-```html
-<link href="node_modules/@angular/material/prebuilt-themes/indigo-pink.css" rel="stylesheet">
+[sass-maps]: https://sass-lang.com/documentation/values/maps
+
+### Predefined palettes
+
+Angular Material offers 19 predefined palettes based on the 2014 version of the Material Design
+spec. See the [Material Design 2014 color palettes][2014-palettes] for a full list.
+
+In addition to hues numbered from zero to 900, the 2014 Material Design palettes each include
+distinct _accent_ hues numbered as `A100`, `A200`, `A400`, and `A700`.
+
+```scss
+@use '~@angular/material' as mat;
+
+$my-palette: mat.$indigo-palette;
 ```
-The actual path will depend on your server setup.
 
-You can also concatenate the file with the rest of your application's css.
+[2014-palettes]: https://material.io/archive/guidelines/style/color.html#color-color-palette
 
-Finally, if your app's content **is not** placed inside a `mat-sidenav-container` element, you
-need to add the `mat-app-background` class to your wrapper element (for example the `body`). This
-ensures that the proper theme background is applied to your page.
+## Themes
 
-## Defining a custom theme
+A **theme** is a collection of color and typography options. Each theme includes three palettes that
+determine component colors:
 
-When you want more customization than a pre-built theme offers, you can create your own theme file.
+* A "primary" palette for the color that appears most frequently throughout your application
+* An "accent", or _secondary_, palette used to selectively highlight key parts of your UI
+* A "warn" palette used for warnings and error states
 
-A custom theme file does two things:
-1. Imports the `mat-core()` Sass mixin. This includes all common styles that are used by multiple
-components. **This should only be included once in your application.** If this mixin is included
-multiple times, your application will end up with multiple copies of these common styles.
-2. Defines a **theme** data structure as the composition of configurations for the individual
-theming systems (`color` and `typography`). This object can be created with either the
-`mat-light-theme` function or the `mat-dark-theme` function. The output of this function is then
-passed to the `angular-material-theme` mixin, which will output all the corresponding styles
-for the theme.
+You can include the CSS styles for a theme in your application in one of two ways: by defining a
+custom theme with Sass, or by importing a pre-built theme CSS file.
 
+### Custom themes with Sass
+
+A **theme file** is a Sass file that calls Angular Material Sass mixins to output color and
+typography CSS styles.
+
+#### The `core` mixin
+
+Angular Material defines a mixin named `core` that includes prerequisite styles for common
```

Comment:
for some reason the `core` mixin takes a typography config. that seems to conflict with the "only include it once" advice, since you could have 2 different themes with different typography

Reply:
I will probably mention this (passing a config) in the typography guide. However, I'd consider it an anti-pattern to define multiple typography settings within an app so this guidance would stand.